extensions/chat-with-content/app.py

```diff
@@ -0,0 +1,304 @@
+            ui.h2("LLM API", class_="setup-section-title"),
+            ui.div(
+                ui.HTML(
+                    "This app requires the <code>CHATLAS_CHAT_PROVIDER</code> and <code>CHATLAS_CHAT_ARGS</code> environment variables to be "
+                    "set along with an LLM API Key in the content access panel. Please set them in your environment before running the app. "
+                    '<a href="https://posit-dev.github.io/chatlas/reference/ChatAuto.html" class="setup-link">See the documentation for more details.</a>'
+                ),
+                class_="setup-description",
+            ),
+            ui.h3("Example for OpenAI API", class_="setup-section-title"),
+            ui.pre(
+                """CHATLAS_CHAT_PROVIDER = "openai"
+CHATLAS_CHAT_ARGS = {"model": "gpt-4o"}
+OPENAI_API_KEY = "<key>" """,
+                class_="setup-code-block",
+            ),
+            ui.h2("Connect Visitor API Key", class_="setup-section-title"),
+            ui.div(
+                "Before you are able to use this app, you need to add a Connect Visitor API Key integration in the access panel.",
+                class_="setup-description",
+            ),
+            class_="setup-card",
+        ),
+        class_="setup-container",
+    ),
+    fillable_mobile=True,
+    fillable=True,
+)
+
+app_ui = ui.page_sidebar(
+    # Sidebar with content selector and chat
+    ui.sidebar(
+        ui.panel_title("Chat with content"),
+        ui.p(
+            "Use this app to select content and ask questions about it. It currently supports static/rendered content."
+        ),
+        ui.input_selectize("content_selection", "", choices=[], width="100%"),
+        ui.output_ui("view_content"),
+        ui.chat_ui(
+            "chat",
+            placeholder="Type your question here...",
+            width="100%",
+        ),
+        width="33%",
+        style="height: 100vh; overflow-y: auto;",
+    ),
+    # Main panel with iframe
+    ui.tags.iframe(
+        id="content_frame",
+        src="about:blank",
+        width="100%",
+        height="100%",
+        style="border: none;",
+    ),
+    # Add JavaScript to handle iframe updates and content extraction
+    ui.tags.script("""
+        window.Shiny.addCustomMessageHandler('update-iframe', function(message) {
+            var iframe = document.getElementById('content_frame');
+            iframe.src = message.url;
+
+            iframe.onload = function() {
+                var iframeDoc = iframe.contentWindow.document;
+                var content = iframeDoc.documentElement.outerHTML;
+                Shiny.setInputValue('iframe_content', content);
+            };
+        });
+    """),
+    fillable=True,
+)
+
+screen_ui = ui.page_output("screen")
+
+CHATLAS_CHAT_PROVIDER = os.getenv("CHATLAS_CHAT_PROVIDER")
+CHATLAS_CHAT_ARGS = os.getenv("CHATLAS_CHAT_ARGS")
+IS_INTERNAL = "connect.posit.it" in os.getenv("CONNECT_SERVER", "")
```

Comment:
I wonder if we should be making the distinction on what chat provider to use on something else - something we can tackle in the future surely.